### DISCUSSION TITLE
Question] The display is normal on the edit screen, but on the article screen it is a single line. The indentation is also wrong. Why?

### DISCUSSION THREAD

**Discussion by ghost** [KevinBatdorf/code-block-pro#347] [Q&A] [answered]
Nice to meet you and hello. I am not very experienced in this kind of posting, so I apologize if I am using it incorrectly.

I am asking this question here because I could not solve the problem by myself.

First of all, I am Japanese and cannot speak any English, so I am using [DeepL translation.](https://www.deepl.com/ja/translator) I apologize if there are any strange English words.

I have installed Code Block Pro on my WordPress since yesterday. It's great.

However, I don't know if it is a compatibility with my theme ([Luxeritas](https://thk.kanzae.net/wp/)) or if there is something wrong with the settings of Code Block Pro, but I can't help but see a single line in the actual post.
![2024年12月09日12時06分43秒](https://github.com/user-attachments/assets/ca4a5d4e-1194-469b-9be5-204792f89657)

I have posted an actual sample article below.
▶(The issue has been resolved and the sample article has been removed and is no longer available.)

It is displayed (formatted) correctly in the article edit screen. For some reason, when I actually publish the article or check it in the draft preview, it becomes a single line.

I checked the developer tools and thought that perhaps adding some CSS to the code tag in div.wp-block-kevinbatdorf-code-block-pro would solve the problem. Since I am a beginner, I looked on the Internet and found the following
▶[https://gist.github.com/KevinBatdorf/1c365af7b799c142ed3cd91a51329f4c](https://gist.github.com/KevinBatdorf/1c365af7b799c142ed3cd91a51329f4c)

I introduced it as a trial and it now breaks lines in the article correctly.(In the attached image, I'm writing to Stylebot, but I'd like to see how it goes a bit more before writing to the WordPress style editor)
![2024年12月09日12時09分52秒](https://github.com/user-attachments/assets/862879c8-5ca7-4010-9b82-7f036852be11)

I thought this was a relief, but there was still a problem: I don't know if it was the introduction of CSS or if there was another cause, but the CSS indentation (Tab) does not seem to display correctly.

Specifically, “pointer-events: none !important;” and “color: inherit !important;” in the sample article are correctly indented (2) on the edit screen, but in the actual article, they are attached to the line numbers and not indented. However, the actual article is attached to the line number and is not indented.

On the other hand, “&>span” is indented for some reason.
![2024年12月09日12時12分14秒 (1)](https://github.com/user-attachments/assets/ed020b2c-3315-4105-9a97-0da6ecf4ea34)

Is this due to applying CSS such as word-break or white-space?

I looked at the developer tools and it is not indented in the first place, so I have a feeling that it is not a CSS issue, but more likely another cause. Is it my theme, a plugin, Code Block Pro settings...?
![2024年12月09日12時17分38秒](https://github.com/user-attachments/assets/50b195ed-2cd3-4944-9e29-91c68387edbf)

That's all the questions I have.

In summary, my first question is: Is it correct to apply CSS to the code tag as the cause of the single line and how to deal with it?

My second question is that when I write wrapping CSS, the indentation is not displayed correctly.

Here is what I have tried as a beginner.
Windows 11 (the latest stable version)
Browser is Vivaldi (the display did not change when viewed in other browsers such as Chrome)
WordPress theme is “Luxeritas”.
WordPress version is 6.7.1
All the following plug-ins are used.
▶Ad Inserter
Beaf - Photo Comparison Block
Code Block Pro
▶ConoHa WING Automatic Cache Clear
▶ Contact Form 7
▶ Customize External Links and add Icon
▶ Dashboard Widgets Suite
▶ EWWW Image Optimizer
▶ Flexible Table Block
▶ Hide Block
▶ Phoenix Media Rename
Public Post Preview
Search Regex
▶ SiteGuard WP Plugin
▶ Spam Protect for Contact Form 7
▶ UpdraftPlus - Backup/Restore
▶ WordPress Import Tools
▶WP Multibyte Patch
▶XML Sitemap Generator for Google
▶Yoast Duplicate Post
▶ YYDevelopment - Show Pages URL List

Sorry for the length of this article.
Thank you for your cooperation.

![2024年12月09日12時51分32秒](https://github.com/user-attachments/assets/bc630ab9-37a4-4bf5-aece-8b4776ea5ba5)
![2024年12月09日12時51分42秒](https://github.com/user-attachments/assets/3c78aa12-99d1-48b3-a0ec-2be99b94dd1e)

**Comment by KevinBatdorf** [comment, marked as answer]
Hey, I think this is a optimization problem and something is removing the whitespace. Have you tried to disable all the plugins? That might help narrow it down. Many plugins that do this will have an option to disable some html minification.

**Reply by ghost** [reply]
Good morning, I'm so glad you enjoyed the article. I am really, really glad you replied. It was a little late at night in Japan, so my reply was delayed.

Wow! Thanks for the advice. I had not tried disabling the plugin yet. I apologize for not doing the most basic of basic tasks.

I will now try turning them on and off one by one to make sure. I hope to be able to add more in an hour or so.

I knew that the problem with the single line and the problem with the indentation not displaying properly could be related problems, not separate. That would be a good study.

**Comment by ghost** [comment]
Thank you for your help.
Sorry for the consecutive replies.

It is amazing. I am currently soaring with joy.

As you told me, I tried disabling all the plug-ins I am using except “Code Block Pro” because I thought it might be caused by HTML optimization.
![2024年12月11日10時23分04秒 (1)](https://github.com/user-attachments/assets/8dfa3f3a-09c6-4e82-9a45-238a1d1638a0)

However, the result did not change. Naturally, I opened the site in a browser with no data when checking to make sure it was not affected by browser cache or cookies.
![2024年12月11日10時20分55秒 (1)](https://github.com/user-attachments/assets/ae0a9db6-a6d8-490e-9cb9-7caa9b5d9779)

I was wondering if there was some setting in the WordPress theme that I was using.

I checked and found it! There it is: !!!! My WordPress theme “Luxeritas” has an option to compress CSS and HTML, and stupidly, the HTML compression ratio was set to “high”...
![2024年12月11日10時24分33秒 (1)](https://github.com/user-attachments/assets/f255c449-9858-45bc-bf71-0b3e71d47db7)

I think I set it to “high” because I was ignorant and thought, “I don't know, but I think it will make the website display faster if I compress it. I really regretted it.

I changed it to the “no compression” option with a sense of excitement...

I was so excited that I changed it to the “no compression” option, and... it was so easy that I wondered what all the trouble I had gone through! The one-line display was also resolved, and the indentation was displayed as in the article editing screen. I am really happy.
![2024年12月11日10時42分11秒](https://github.com/user-attachments/assets/29690ea8-32f4-4b5c-9d38-b834541e21c6)

If I hadn't been told about this by the experts, I would have thought it was a problem with Code Block Pro or the CSS I was applying, and I would have thought it was misguided.

Thank you so much !!!!!!

**Reply by KevinBatdorf** [reply]
That's great news and thanks for coming here so I can help. I'm sure your message will help someone with the same theme that experiences the same issue another day :)

Take care, and if you have any other issues let me know.
